src/stream.rs

```diff
@@ -0,0 +1,63 @@
+// TODO: move this module to deflate64 crate
+
+use crate::InflaterManaged;
+use std::io;
+use std::io::{BufRead, BufReader, Read};
+
+/// The reader the decompresses deflate64 from another BufRead.
+pub struct Deflate64Decoder<R> {
+    inner: R,
+    inflater: Box<InflaterManaged>,
```

Comment:
Is there any reason to box `InflaterManaged`?

IMO boxing should be avoid to achieve zero-cost, the users can choose to box it themselves if they deem it to be too large.

Reply:
`InflaterManaged` is very big struct because of internal buffer and I think instance of Read is likely to be allocated on stack so I choose to use `Box<InflaterManaged>`.

flate2 does similar thing because of same reason. 

https://github.com/rust-lang/flate2-rs/blob/b1e993aa57a0af20a1de7c54cc23937fe46fdae5/src/deflate/bufread.rs#L156
https://github.com/rust-lang/flate2-rs/blob/b1e993aa57a0af20a1de7c54cc23937fe46fdae5/src/mem.rs#L39
https://github.com/rust-lang/flate2-rs/blob/b1e993aa57a0af20a1de7c54cc23937fe46fdae5/src/ffi/rust.rs#L38

Reply:
It makes sense then given its size, although it could be possible that users also box `Deflate64Decoder`.